tests/ragged_inference/test_triton_v2_ragged_qk_dotprod.py

```diff
@@ -109,9 +112,77 @@ def test_ragged_qk_dotprod_multiple_seqs():
     torch_scores = scores_via_qk_dotprod(query, key)
     scores = ragged_qk_dotprod(query, key, lut)
 
-    for seq_idx, (n_ctx_q, n_ctx_k) in enumerate(zip(key.n_ctx_per_seq, query.n_ctx_per_seq)):
-        print(f'Checking {seq_idx=}')
-        assert_eq(torch_scores[seq_idx, :n_ctx_q, :n_ctx_k], scores[seq_idx, :n_ctx_q, :n_ctx_k])
+    for seq_idx, (n_ctx_q, n_ctx_k) in enumerate(
+        zip(key.n_ctx_per_seq, query.n_ctx_per_seq)
+    ):
+        print(f"Checking {seq_idx=}")
+        assert_eq(
+            torch_scores[seq_idx, :n_ctx_q, :n_ctx_k],
+            scores[seq_idx, :n_ctx_q, :n_ctx_k],
+        )
+
+
+def test_ragged_qk_dotprod_multiple_seqs_perf():
+    n_q_ctx = 5
+    n_seqs = 50
+    d_head = 256
+    n_k_ctx = 8000
+    n_iters = 10
+
+    query = RaggedActivations.from_list(
+        [
+            _make_seq_arange(n_ctx=n_q_ctx, start_value=0, d_head=d_head)
+            for _ in range(n_seqs)
+        ]
+    )
+    key = RaggedActivations.from_list(
+        [
+            _make_seq_arange(n_ctx=n_k_ctx, start_value=0, d_head=d_head)
+            for _ in range(n_seqs)
+        ]
+    )
+
+    lut = RaggedQkPidLookupTable.from_query_and_key_tokens_per_seq(
+        n_ctx_q_per_seq=query.n_ctx_per_seq,
+        n_ctx_k_per_seq=key.n_ctx_per_seq,
+    )
+
+    for _ in range(3):
+        out = ragged_qk_dotprod(query, key, lut)
+
+    torch.cuda.synchronize()
+    started_at = time.time()
+    for _ in range(n_iters):
+        out = ragged_qk_dotprod(query, key, lut)
```

Comment:
nit, but there could be caching/warm issues here (would not change the perf that much, but FYI). Triton exposes a great too for this, we use it in most benchmarks, see [here](https://github.com/facebookresearch/xformers/blob/main/xformers/benchmarks/benchmark_triton_dropout.py#L90) for instance (it wipes the GPU cache in between iterations to cover that if my memory serves me well)